vendor/bat-native-ads/src/bat/ads/internal/search_providers.cc

```diff
@@ -45,4 +66,18 @@ bool SearchProviders::IsSearchEngine(const std::string& url) {
   return is_a_search;
 }
 
+bool SearchProviders::HasResultsPagesForUrl(
+    const std::string& url,
+    const SearchProviderInfo& search_provider) {
+  size_t index = search_provider.search_template.find('{');
+  std::string substring = search_provider.search_template.substr(0, index);
+  size_t href_index = url.find(substring);
+
+  if (index == std::string::npos || href_index == std::string::npos) {
```

Comment:
We're checking down here if index is an invalid value, but we've already used it up above. I think it'd be better to just split this up into two individual if statements and test them as soon as they're set.

Reply:
fixed, thanks